### PR TITLE
Remove ProgrammingEvaluation model.

### LINKED ISSUE #2345: Remove Course::Assessment::ProgrammingEvaluation model
Also removed the associated controllers, routes and everything else.

https://github.com/Coursemology/coursemology2/blob/master/app/models/course/assessment/programming_evaluation.rb

This was needed to "publish" evaluations for the evaluators to pick up.

#2308 makes this unnecessary as external evaluators are no longer used.

### PR DESCRIPTION
**Pull request by fonglh**
For `course_assessment_programming_evaluations` model:

1. Remove model, controller and factory.
1. Clean up references to the model in other code.
1. Drop database table.
1. Remove routes.

Fixes #2345.

### CODE REVIEW

File: spec/support/stubs/course/assessment/stubbed_programming_evaluation_service.rb [hunk 1]
```diff
@@ -3,8 +3,11 @@ module Course::Assessment::StubbedProgrammingEvaluationService
   private
 
   def evaluate_in_container
-    attributes = FactoryGirl.attributes_for(:course_assessment_programming_evaluation, :completed).
-                   slice(:stdout, :stderr, :test_report, :exit_code)
+    attributes = { stdout: '',
+                   stderr: '',
+                   test_report: File.read(Rails.root.join('spec/fixtures/course/' \
+                                                          'programming_multiple_test_suite_report.xml')),
```
Comment (fonglh) [inline spec/support/stubs/course/assessment/stubbed_programming_evaluation_service.rb]: Won't fix. Splitting the filename would be bad for readability.

File: db/migrate/20170706030838_drop_course_assessment_programming_evaluations.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class DropCourseAssessmentProgrammingEvaluations < ActiveRecord::Migration
```
Comment (fonglh) [inline db/migrate/20170706030838_drop_course_assessment_programming_evaluations.rb]: Won't fix, none of the other migrations have them.

File: db/migrate/20170706030838_drop_course_assessment_programming_evaluations.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+  def change
+    drop_table :course_assessment_programming_evaluations
```
Comment (fonglh) [inline db/migrate/20170706030838_drop_course_assessment_programming_evaluations.rb]: That's ok, it only contains ephemeral data anyway.

**Comment by weiqingtoh** [comment]
Do we also need to remove the auto-grader user?

**Comment by fonglh** [comment]
Ah yes we should.

**Comment by weiqingtoh** [comment]
Also need to remove this -> https://github.com/Coursemology/coursemology2/blob/master/db/seeds/development.rb#L12-L19

**Comment by weiqingtoh** [comment]
The last part to all of this is to figure out whether we should also remove the auto-grader user in production, and if so, how. 

Removing it in production by adding a delete autograder user might be disallowed due to some FK constraint (it might be the creator/updater or some other models). Should we also actively replace that with say the system user or something?

**Comment by fonglh** [comment]
Possible FK constraints are in the `instances` and `user_emails` tables.

This was confirmed by checking the schema file for tables with a `user_id` column referencing the `users` table. There are 10 matches. For each match, `SELECT * FROM <table> WHERE user_id=<autograder_user_id>` shows that only those 2 tables are affected.

**Comment by weiqingtoh** [comment]
Ok. I think no issues cause there is `dependent: destroy` declared on both associations. How do you prefer to delete the user - through a migration or manually in production?

**Comment by fonglh** [comment]
I'll update the migration, better to have it all in 1 place.

**Comment by weiqingtoh** [comment]
Can rebase? The rest LGTM

**Comment by weiqingtoh** [comment]
1 more in `lib/tasks/coursemology/seed.rake`. Maybe do a global search to just see where else?

**Pull request merged**